README.md

```diff
@@ -72,7 +72,7 @@ $obj = $(Get-WMIObject Win32_UserAccount -Filter "Name='Guest'")
 $obj.Rename("OtherGuest")
 ~~~
 
-This has the added benefit of not requiring escaping '$' in the PowerShell code.
+This has the added benefit of not requiring escaping '$' in the PowerShell code. Note that templates need to have DOS linefeeds or they will not work as expected. One tool for converting UNIX linefeeds to DOS linefeeds is [unix2dos](http://freecode.com/projects/dos2unix).
```

Comment:
We should just fix the provider. If `content` contains the sequence `\r\n`, then open the tempfile in `binary` mode, preserving the existing line endings. Otherwise, `content` doesn't contain any line endings or is using `\n`, and we should open the tempfile in `text` mode, ensuring the resulting script has `\r\n`.